src/ucp/wireup/wireup_cm.c

```diff
@@ -266,10 +271,14 @@ ucp_cm_ep_priv_data_pack(ucp_ep_h ep, const ucp_tl_bitmap_t *tl_bitmap,
     if (status != UCS_OK) {
         goto err;
     }
+    priv_data_length = ucp_cm_priv_data_length(ucp_addr_size);
+    wireup_ep        = ucp_ep_get_cm_wireup_ep(ep);
+    if (priv_data_pack && (wireup_ep->client_id != 0)) {
```

Comment:
maybe no need to pass priv_data_pack and just check by `wireup_ep->client_id != 0` ?